features/0434-outofband/README.md

```diff
@@ -0,0 +1,417 @@
+
+The *receiver* state machine is as follows:
+
+[![receiver state machine google doc](state-machine-receiver.png)](https://docs.google.com/spreadsheets/d/1o1szGCMx6FzrxfAKV76IEYmeKRJqvV3gd1r3Qfv1Q9o/edit#gid=10874521)
+
+Worth noting is the first event of the `done` state, where the receiver may receive the message multiple times. This represents, for example, an agent returning to the same website and being greeted with the same QR code each time.
+
+### Messages
+
+There are two messages in the out of band protocol. They are both sent by the *sender*. Only one of the message types is ever used in an instance of the protocol.
+
+#### Message Type: `https://didcomm.org/outofband/VER/invitation`
+
+```jsonc
+{
+  "@type": "https://didcomm.org/outofband/VER/invitation",
+  "@id": "<id used for context as pthid>",
+  "label": "Faber College",
+  "protocols": [
+      "https://didcomm.org/did-exchange/1.0",
+      "https://didcomm.org/connections/1.0"
+      ]
+  "service": ["did:sov:LjgpST2rjsoxYegQDRm7EL"]
+}
+```
+
+The items in the message are:
+
+- `@type` - the DIDComm message type
+- `@id` - the unique ID of the message. The ID should be used as the **parent** thread ID (`pthid`) for the response message, rather than the more common thread ID (`thid`) of the response message. This enables multiple uses of a single `invitation` message.
+- `label` - a self-attested string that the receiver may want to display to the user.
+- `protocols` - array of protocols in the order of preference of the sender that the receiver can use in responding to the message. As implied by the message type of invitation, these are not arbitrary protocols but rather protocols that result in the establishment of a connection.
+- `service` - an item that is the equivalent of the service block of a DIDDoc that the receiver is to use in responding to the message. Additional details below.
+
+While the _receiver_ is expected to respond with an initiating message from the chosen protocol using a specified service, the receiver may respond by reusing an existing connection. Specifically, if a connection they have was created from an out of band `invitation` from the same public DID of a new `invitation`, the receiver **SHOULD** use the existing connection in responding to the `invitation`. The selected `protocol` being used by the receiver may have a specific message type to use for this purpose.
+
+> **To Do**: Can we expand this to be able to reuse an inline `service` entry?
+> **To Do**: Update the `did-exchange` (and perhaps `connection`) protocol to have a `continue` message to be used for this purpose.
+
+#### Message Type: `https://didcomm.org/outofband/VER/request`
+
+```jsonc
+{
+  "@type": "https://didcomm.org/outofband/VER/request",
```

Comment:
```suggestion
  "@type": "https://didcomm.org/outofband/1.0/request",
```

Reply:
We've talked about only putting the version in the header so there is no confusion about what is the current version.

Are you OK with that, or do we include the actual version in the JSON examples?

Reply:
Oh - I wasn't aware of that discussion.

Reply:
Good point. I was kind of testing this out to see what people think.

@dhh1128 @TelegramSam - what do you think of using this approach to version, the current version number is only in the header?

Reply:
After this, we should think about adjusting the protocol template further.  I think it is better, but not perfect.  I definitely didn't like the messages and state sections being in both the tutorial and the reference.  I think we should stick with one, or provide better guidance on what puts in one or the other.

Reply:
I'm okay with a placeholder, but I don't think "VER" in all caps is quite good enough. The reason is that this just looks like another URL segment that happens to be capitalized. Could we use a delimiter that would have to be URL-encoded, so no person could believe that what they're reading is an actual URL? Something like `%VER`, since we know `%` can't appear followed by `VER` in a valid URL?

Reply:
Changed. If we like that, we'll add that to the protocol template.